### PR TITLE
added layout

### PR DESCRIPTION
**Pull request by dgempiuc**
layout field added and encode method modified by using the layout while adding the messages to the log file.

### CODE REVIEW

**Comment by fabian-pena** [comment]
Any news on this PR? It would be very useful to be able to configure the layout to customize the message.

File: logback-ecs-encoder/src/main/java/co/elastic/logging/logback/EcsEncoder.java [hunk 1]
```diff
@@ -54,6 +55,7 @@ public class EcsEncoder extends EncoderBase<ILoggingEvent> {
     private boolean includeOrigin;
     private final List<AdditionalField> additionalFields = new ArrayList<AdditionalField>();
     private OutputStream os;
+    protected Layout<ILoggingEvent> layout;
```
Comment (eyalkoren) [inline logback-ecs-encoder/src/main/java/co/elastic/logging/logback/EcsEncoder.java:58]:
```suggestion
    protected Layout<ILoggingEvent> messageLayout;
```

File: logback-ecs-encoder/src/main/java/co/elastic/logging/logback/EcsEncoder.java [hunk 2]
```diff
@@ -136,6 +138,14 @@ public byte[] encode(ILoggingEvent event) {
         return builder.toString().getBytes(UTF_8);
     }
 
+    private void serializeMessage(ILoggingEvent event, StringBuilder builder) {
+        if(layout == null) {
+            EcsJsonSerializer.serializeFormattedMessage(builder, event.getFormattedMessage());
+        } else {
+            EcsJsonSerializer.serializeFormattedMessage(builder, this.layout.doLayout(event));
```
Comment (eyalkoren) [inline logback-ecs-encoder/src/main/java/co/elastic/logging/logback/EcsEncoder.java:145]:
```suggestion
            EcsJsonSerializer.serializeFormattedMessage(builder, this.messageLayout.doLayout(event));
```

File: logback-ecs-encoder/src/main/java/co/elastic/logging/logback/EcsEncoder.java [hunk 3]
```diff
@@ -198,4 +208,11 @@ public void setEventDataset(String eventDataset) {
     public void setThrowableConverter(ThrowableHandlingConverter throwableConverter) {
         this.throwableConverter = throwableConverter;
     }
+    public Layout<ILoggingEvent> getLayout() {
+        return this.layout;
+    }
+
```
Comment (eyalkoren) [inline logback-ecs-encoder/src/main/java/co/elastic/logging/logback/EcsEncoder.java:214]:
```suggestion
```
No need for a getter

File: logback-ecs-encoder/src/main/java/co/elastic/logging/logback/EcsEncoder.java [hunk 4]
```diff
@@ -198,4 +208,11 @@ public void setEventDataset(String eventDataset) {
...
+    public void setLayout(Layout<ILoggingEvent> layout) {
+        this.layout = layout;
+    }
```
Comment (eyalkoren) [inline logback-ecs-encoder/src/main/java/co/elastic/logging/logback/EcsEncoder.java:217]:
```suggestion
    public void setMessageLayout(Layout<ILoggingEvent> messageLayout) {
        this.messageLayout = messageLayout;
    }
```
Add a javadoc saying this `Layout` will be applied specifically to format the `message` field based on the logging event.

**Comment by felixbarny** [comment]
> @felixbarny any reason you see for not adding it?

No objections but it would be interesting to know about the use case this solves. Why would you want to use a layout within the `message` field?

**Comment by rdifrango** [comment]
@felixbarny the reason for needing this is covered by #144, basically being able to mask fields in environments where you are handling sensitive data would be good to have in this encoder.

If @dgempiuc can't update this PR to get it merged, I would be willing to contribute this as well.

**Comment by felixbarny** [comment]
> If @dgempiuc can't update this PR to get it merged, I would be willing to contribute this as well.

SGTM. But please be sure to take his commits as the foundation so that he still appears as a co-author of your PR.

**Comment by maxiking445** [comment]
> @felixbarny the reason for needing this is covered by #144, basically being able to mask fields in environments where you are handling sensitive data would be good to have in this encoder.
> 
> If @dgempiuc can't update this PR to get it merged, I would be willing to contribute this as well.

Any news here? @rdifrango

**Comment by rdifrango** [comment]
> 

Unfortunately, I haven't had time to get back to it...I had it staged and started the proposed changes but haven't yet finished them :(

**Comment by rdifrango** [comment]
@maxiking445 I'm working through the process now to contribute this back.  It'll likely be in the form of a new PR with the commits from @dgempiuc as the base.

**Comment by rdifrango** [comment]
Replaced by #220

**Comment by rdifrango** [comment]
@JonasKunz now that #220 is merged we should go ahead and close this one as it supersedes it.

**Pull request closed without merging**